src/protocol/errors.rs

```diff
@@ -49,3 +96,27 @@ pub enum InvalidSnapshotError {
     #[error("Value error {0}")]
     ValueError(String),
 }
+
+/// Represents the outer-level, user-facing errors of the tycho-simulation package.
+///
+/// `TychoSimulationError` encompasses all possible errors that can occur in the package,
+/// wrapping lower-level errors in a user-friendly way for easier handling and display.
+#[derive(Error, Debug)]
+pub enum TychoSimulationError {
+    #[error("VM simulation error: {0}")]
+    VMError(VMError),
+    #[error("Native simulation error: {0}")]
+    NativeSimulationError(NativeSimulationError),
+}
```

Comment:
Why are we differentiating between these errors (vm vs native) - I don't think we should make a differentiation here as this is an implementation specific detail that shouldn't leak into the interface! 

I think we should merge all of the cases this into a single error struct. Let's also get rid of the `...ErrorKind` enum.